torch/_refs/__init__.py

```diff
@@ -2877,6 +2878,101 @@ def vsplit(
     return tensor_split(a, split_sizes, 0)
 
 
+# See maybe_wrap_dim in core
+def _maybe_wrap_dim(
```

Comment:
there's already `canonicalize_dim` in prims_common. It doesn't handle scalar, but you can add that handling at the call site (as most other refs do)